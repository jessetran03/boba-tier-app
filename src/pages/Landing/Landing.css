.landing-cities li {
  display: inline;
  margin-right: 20px;
}

.landing-cities button {
  font-size: 15px;
  font-weight: 600;
  padding: 10px;
  border-radius: 20px;
}

.landing-cities button:hover {
  cursor: pointer;
  background-color: #CCCCCC;
}