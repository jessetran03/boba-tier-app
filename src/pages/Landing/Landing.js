import React, { Component } from 'react'
import drink1 from '../../images/boba-drink-1.jpg'
import drink2 from '../../images/boba-drink-2.jpg'
import drink3 from '../../images/boba-drink-3.jpg'
import config from '../../config'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import TokenService from '../../services/token-service'

import './Landing.css'

export default class Landing extends Component {
  state = {
    error: null,
    loading: false,
  }

  handleLoginSuccess = () => {
    const { location, history } = this.props
    const destination = (location.state || {}).from || '/'
    history.push(destination)
  }

  handleDemo = e => {
    e.preventDefault()
    this.setState({ 
      error: null,
      loading: true, 
    })
    const login = {
      user_name: 'Boba Guest',
      password: 'BobaGuest1!',
    }
    fetch(`${config.API_ENDPOINT}/auth/login`, {
      method: 'POST',
      headers: {
        'content-type': 'application/json'
      },
      body: JSON.stringify(login),
    })
      .then(res => {
        if (!res.ok)
          return res.json().then(e => Promise.reject(e))
        return res.json()
      })
      .then(res => {
        this.setState({ loading: false })
        TokenService.saveAuthToken(res.authToken)
        this.handleLoginSuccess()
      })
      .catch(res => {
        this.setState({ 
          error: res.error ,
          loading: false,
        })
      })
  };

  renderLoggedOut() {
    const loading = this.state.loading
    return (
      <section className='landing-page'>
        <section className="landing">
          <h2>Rate the Best Boba Tea in Town!</h2>
          <p className='boba-description'>
            Boba Tier lets you rate all the drinks in boba shops by city.
            Along with keeping track of all the drinks you previously rated,
            you will also be able to see which drinks are the highest rated for that city.
            You can also leave comments on the shop pages expressing your thoughts about their drinks.
            Get started by using the navigation pane or click on demo to try it out as a guest!
          </p>
          <button onClick={this.handleDemo} className='demo'>
            Demo
          </button>
          <div>
            {loading && <FontAwesomeIcon className='spinner' icon='spinner' spin />}
          </div>
          <p className='city-notice'>
            Boba Tier is currently available for shops in Houston, TX. More cities will be added in the future!
          </p>
        </section>
        <section className='landing-info'>
          <img src={drink2} alt='boba drink 2' className='landing-image' />
          <img src={drink1} alt='boba drink 1' className='landing-image' />
          <img src={drink3} alt='boba drink 3' className='landing-image' />
        </section>
      </section>
    )
  }

  renderLoggedIn() {
    return (
      <section className='landing-page'>
        <section className="landing">
          <h2>Rate the Best Boba Tea in Town!</h2>
          <p className='boba-description'>
            Boba Tier lets you rate all the drinks in boba shops by city.
            Along with keeping track of all the drinks you previously rated,
            you will also be able to see which drinks are the highest rated for that city.
            You can also leave comments on the shop pages expressing your thoughts about their drinks.
            Get started by using the navigation pane!

          </p>
          <p className='city-notice'>
            Boba Tier is currently available for shops in Houston, TX. More cities will be added in the future!
          </p>
        </section>
        <section className='landing-info'>
          <img src={drink2} alt='boba drink 2' className='landing-image' />
          <img src={drink1} alt='boba drink 1' className='landing-image' />
          <img src={drink3} alt='boba drink 3' className='landing-image' />
        </section>
      </section>
    )
  }

  render() {
    return (
      <>
        {TokenService.hasAuthToken()
          ? this.renderLoggedIn()
          : this.renderLoggedOut()
        }
      </>
    )
  }
}