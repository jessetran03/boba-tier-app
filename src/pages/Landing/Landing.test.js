import React from 'react'
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import Landing from './Landing'
import renderer from 'react-test-renderer'

it('renders Landing without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <BrowserRouter>
      <Landing />
    </BrowserRouter>,
    div
  )
  ReactDOM.unmountComponentAtNode(div)
})

it('render Landing without data', () => {
  const wrapper = renderer.create(
    <BrowserRouter>
      <Landing />
    </BrowserRouter>
  )
    .toJSON();
  expect(wrapper).toMatchSnapshot()
})