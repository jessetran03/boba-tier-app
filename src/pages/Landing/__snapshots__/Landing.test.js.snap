// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Landing without data 1`] = `
<section
  className="landing-page"
>
  <section
    className="landing"
  >
    <h2>
      Rate the Best Boba Tea in Town!
    </h2>
    <p
      className="boba-description"
    >
      Boba Tier lets you rate all the drinks in boba shops by city. Along with keeping track of all the drinks you previously rated, you will also be able to see which drinks are the highest rated for that city. You can also leave comments on the shop pages expressing your thoughts about their drinks. Get started by using the navigation pane or click on demo to try it out as a guest!
    </p>
    <button
      className="demo"
      onClick={[Function]}
    >
      Demo
    </button>
    <div />
    <p
      className="city-notice"
    >
      Boba Tier is currently available for shops in Houston, TX. More cities will be added in the future!
    </p>
  </section>
  <section
    className="landing-info"
  >
    <img
      alt="boba drink 2"
      className="landing-image"
      src="boba-drink-2.jpg"
    />
    <img
      alt="boba drink 1"
      className="landing-image"
      src="boba-drink-1.jpg"
    />
    <img
      alt="boba drink 3"
      className="landing-image"
      src="boba-drink-3.jpg"
    />
  </section>
</section>
`;
